galsim/phase_psf.py

```diff
@@ -0,0 +1,1017 @@
+                         the meta-pupil defined by the wind velocity and exposure time.  Note that
+                         the screen will have periodic boundary conditions, so it's technically
+                         possible to use a smaller sized screen than technically necessary, though
+                         this may introduce artifacts into PSFs or PSF correlations functions.
+                         Note that screen_size may be tweaked by the initializer to ensure
+                         screen_size is a multiple of screen_scale.
+    @param screen_scale  Physical pixel scale of phase screen in meters.  A fraction of the Fried
+                         parameter is usually sufficiently small, but users should test the effects
+                         of this parameter to ensure robust results. [Default: half of r0_500]
+    @param altitude      Altitude of phase screen in km.  This is with respect to the telescope, not
+                         sea-level.  [Default: 0.0]
+    @param time_step     Interval to use when advancing the screen in time in seconds.
+                         [Default: 0.03]
+    @param r0_500        Fried parameter setting the amplitude of turbulence; contributes to "size"
+                         of the resulting atmospheric PSF.  Specified at wavelength 500 nm, in units
+                         of meters.  [Default: 0.2]
+    @param L0_inv        Inverse outer scale in inverse meters, beyond which turbulence power
+                         spectrum smoothly approaches zero.  [Default: 1./25]
+    @param vx            x-component wind velocity in meters/second.  [Default: 0.]
+    @param vy            y-component wind velocity in meters/second.  [Default: 0.]
+    @param rng           Random number generator as a galsim.BaseDeviate().  If None, then use the
+                         clock time or system entropy to seed a new generator.  [Default: None]
+    """
+    def __init__(self, screen_size, screen_scale=None, altitude=0.0, time_step=0.03,
+                 r0_500=0.2, L0_inv=1./25.0, vx=0.0, vy=0.0, rng=None):
+
+        if screen_scale is None:
+            screen_scale = 0.5 * r0_500
+        super(AtmosphericScreen, self).__init__(screen_size, screen_scale, altitude)
+
+        if rng is None:
+            rng = galsim.BaseDeviate()
+        self.rng = rng
+        self.orig_rng = copy.deepcopy(rng)
+
+        self.time_step = time_step
+        self.altitude = altitude
+        self.r0_500 = r0_500
+        self.L0_inv = L0_inv
+        self.vx = vx
+        self.vy = vy
+
+    def advance_by(self, dt):
+        """Advance phase screen by specified amount of time.
+
+        @param dt  Amount of time in seconds by which to update the screen.
+        @returns   The actual amount of time updated, which will differ from `dt` when `dt` is not a
+                   multiple of self.time_step.
+        """
+        _nstep = int(np.ceil(dt/self.time_step))
```

Comment:
Maybe round is more appropriate than ceil?  Although maybe with a minimum of nstep=1.  I'm not sure if rounding is likely to be an issue here, but I'm worried that people will think they are advancing n * time_step, and rounding will result in them being advanced by (n+1) * time_step.